osu.Game.Rulesets.Mania/UI/ManiaStage.cs

```diff
@@ -150,6 +150,41 @@ public ManiaStage(int firstColumnIndex, StageDefinition definition, ref ManiaAct
             }, true);
         }
 
+        private ISkin currentSkin;
+
+        [BackgroundDependencyLoader]
+        private void load(ISkinSource skin)
+        {
+            currentSkin = skin;
+            skin.SourceChanged += onSkinChanged;
+
+            onSkinChanged();
+        }
+
+        private void onSkinChanged()
+        {
+            foreach (var col in columnFlow)
+            {
+                if (col.Index > 0)
+                {
+                    float spacing = currentSkin.GetConfig<LegacyManiaSkinConfigurationLookup, float>(
+                                                   new LegacyManiaSkinConfigurationLookup(Columns.Count, LegacyManiaSkinConfigurationLookups.ColumnSpacing, col.Index - 1))
+                                               ?.Value ?? COLUMN_SPACING;
+
+                    col.Margin = new MarginPadding { Left = spacing };
+                }
+
+                float? width = currentSkin.GetConfig<LegacyManiaSkinConfigurationLookup, float>(
+                                              new LegacyManiaSkinConfigurationLookup(Columns.Count, LegacyManiaSkinConfigurationLookups.ColumnWidth, col.Index))
+                                          ?.Value;
+
+                if (width == null)
+                    col.Width = col.IsSpecial ? Column.SPECIAL_COLUMN_WIDTH : Column.COLUMN_WIDTH;
```

Comment:
Most skins assume that the column width is 30 instead of the lazer values. Might be another PR imo.

Reply:
Check the line below this one, where I'm using the skin width.

Reply:
this is only ever hit by the default skin, correct?

i think we may want to re-think this lookup process. at very least the configuration lookup type should be renamed from "legacy" if we are using it in a mainstream class (`ManiaStage`)

Reply:
Yeah there's some configs that are 100% legacy, and others that probably aren't going to be, such as column width and spacing.

I didn't split it out because I'm not sure I want to implement 3 ways of retrieving mania skin stuff though: `ManiaSkinComponent`, `LegacyManiaSkinConfigurationLookup`, `ManiaSkinConfigurationLookup`, all before we have a system for dealing with drawable configs.